cmake/Modules/thread-local.cmake

```diff
@@ -0,0 +1,25 @@
+include(CheckCXXSourceCompiles)
+
+CHECK_CXX_SOURCE_COMPILES("
+static thread_local int tls;
+int main(void)
+{
+  return 0;
+}" HAVE_THREAD_LOCAL)
```

Comment:
`thread_local` is standard c++11 and since we require a conforming compiler, why not use that without any extra checks and alternatives?